AnimationSystem/AnimationSystem.cc

```diff
@@ -0,0 +1,1074 @@
+          else if (boneName == "Right_indexFinger3") boneIndexes.Right_indexFinger3 = i;
+          else if (boneName == "Right_ringFinger1") boneIndexes.Right_ringFinger1 = i;
+          else if (boneName == "Right_ringFinger2") boneIndexes.Right_ringFinger2 = i;
+          else if (boneName == "Right_ringFinger3") boneIndexes.Right_ringFinger3 = i;
+          else if (boneName == "Right_littleFinger1") boneIndexes.Right_littleFinger1 = i;
+          else if (boneName == "Right_littleFinger2") boneIndexes.Right_littleFinger2 = i;
+          else if (boneName == "Right_littleFinger3") boneIndexes.Right_littleFinger3 = i;
+          else if (boneName == "Right_leg") boneIndexes.Right_leg = i;
+          else if (boneName == "Right_knee") boneIndexes.Right_knee = i;
+          else if (boneName == "Right_ankle") boneIndexes.Right_ankle = i;
+          else if (boneName == "Right_toe") boneIndexes.Right_toe = i;
+          else if (boneName == "Left_leg") boneIndexes.Left_leg = i;
+          else if (boneName == "Left_knee") boneIndexes.Left_knee = i;
+          else if (boneName == "Left_ankle") boneIndexes.Left_ankle = i;
+          else if (boneName == "Left_toe") boneIndexes.Left_toe = i;
+        }
+      }
+
+      // -------------------------------------------------------------------------
+
+      jsonStr += "[";
+      for (unsigned int i = 0; i < declarations.size(); i++) {
+        AnimationGroupDeclaration declaration = declarations[i];
+        std::vector<Animation *> animationGroup;
+
+        jsonStr += "{";
+        jsonStr += "\"name\":";
+        jsonStr += "\"" + declaration.groupName + "\"";
+        jsonStr += ",";
+        jsonStr += "\"index\":";
+        jsonStr += std::to_string(declaration.index);
+        jsonStr += ",";
+        jsonStr += "\"animations\":";
+        jsonStr += "[";
+        for (unsigned int j = 0; j < declaration.animationDeclarations.size(); j++) {
+          AnimationDeclaration animationDeclaration = declaration.animationDeclarations[j];
+          animationGroup.push_back(animationAll[animationDeclaration.fileName]);
+
+          jsonStr += "{";
+          jsonStr += "\"keyName\":";
+          jsonStr += "\"" + animationDeclaration.keyName + "\"";
+          jsonStr += ",";
+          jsonStr += "\"index\":";
+          jsonStr += std::to_string(animationDeclaration.index);
+          jsonStr += ",";
+          jsonStr += "\"fileName\":";
+          jsonStr += "\"" + animationDeclaration.fileName + "\"";
+          jsonStr += "}";
+          if (j != declaration.animationDeclarations.size() - 1) jsonStr += ",";
+        }
+        animationGroups.push_back(animationGroup);
+        jsonStr += "]";
+        jsonStr += "}";
+        if (i != declarations.size() - 1) jsonStr += ",";
+      }
+      jsonStr += "]";
+
+      // -------------------------------------------------------------------------
+
+      defaultSitAnimationIndex = sitAnimationIndexes.Chair;
```

Comment:
These defaults should not exist in C++. They must be computed on the JS side and always passed to C++.

If you don't explicitly pass the index you want, that's an error.

Reply:
https://github.com/webaverse/app-wasm/pull/75/files#diff-bfaaa55a4ec2614935143fb5dabf5097733ba995063b5fec95020a26bf65c7b9R713-R727
Because of now defining these animations infos directly in C++, not in JS.
So I think we can't and don't need compute on JS side then pass to C++.
And then will never pass wrong index.